### ISSUE TITLE
[bug] Artifact preview not working with default minio, and no experiment namespace (standalone install)

### ISSUE THREAD

**Issue by csanadpoda** [kubeflow/pipelines#12224]
### Environment

*  How did you deploy Kubeflow Pipelines (KFP)?
I deployed it on bare metal as part of the Kubeflow AI Reference Platform, version 1.10
*  KFP version: 2.11.0
build version - dev_local

Create an Experiment with a Run from Kubeflow Notebooks. Define some plots to be displayed. Run the experiment, check the results on the UI:

<img width="3420" height="1966" alt="Image" src="https://github.com/user-attachments/assets/e97b7e3f-71c5-49fc-ac8d-cff5bd81cd14" />

<img width="3420" height="1970" alt="Image" src="https://github.com/user-attachments/assets/d2195b36-d1e0-4640-b25b-a6e4ef30fcb6" />

Pipeline/Workspace also not showing:

<img width="3418" height="1958" alt="Image" src="https://github.com/user-attachments/assets/f8c8f563-213a-48e9-88c2-0b11fa22f4d9" />

I think this is the same issue as https://github.com/kubeflow/pipelines/issues/11058 but that was supposedly resolved over a year ago, but I installed the latest regerence platform maybe a month ago and still encounter the issue.

### Expected result

Minio should be able to show the correct plot.

### Materials and Reference

Exact code:
`import uuid
import base64
import kfp
from kfp import dsl
from kfp.dsl import Output, Input, Metrics, Markdown, Model

TRAIN_IMAGE = "jupyter/scipy-notebook:python-3.11"

@dsl.component(base_image=TRAIN_IMAGE)
def train_and_log(epochs: int, metrics: Output[Metrics], loss_plot: Output[Markdown], model: Output[Model]):
    import os, io, joblib, numpy as np, base64, time
    from sklearn.datasets import load_iris
    from sklearn.model_selection import train_test_split
    from sklearn.linear_model import SGDClassifier
    from sklearn.metrics import log_loss, accuracy_score
    X, y = load_iris(return_X_y=True)
    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42, stratify=y)
    clf = SGDClassifier(loss="log_loss", max_iter=1, tol=None, learning_rate="constant", eta0=0.05, random_state=42)
    classes = np.unique(y_train)
    losses = []
    t0 = time.time()
    for i in range(epochs):
        clf.partial_fit(X_train, y_train, classes=classes)
        p = clf.predict_proba(X_train)
        l = float(log_loss(y_train, p, labels=classes))
        losses.append(l)
        mean = (time.time() - t0) / (i + 1)
        eta = mean * (epochs - i - 1)
        print(f"[{i+1}/{epochs}] loss={l:.6f} ETA={eta:.1f}s", flush=True)
    y_pred = clf.predict(X_test)
    acc = float(accuracy_score(y_test, y_pred))
    metrics.log_metric("final-loss", float(losses[-1]))
    metrics.log_metric("accuracy", acc)
    xs = list(range(1, epochs + 1))
    x0, x1 = min(xs), max(xs)
    y0, y1 = min(losses), max(losses) if max(losses) != min(losses) else min(losses) + 1e-6
    w, h, m = 640, 320, 40
    sx = lambda x: m + (w - 2 * m) * (x - x0) / (x1 - x0)
    sy = lambda y: h - m - (h - 2 * m) * (y - y0) / (y1 - y0)
    pts = " ".join(f"{sx(x):.2f},{sy(y):.2f}" for x, y in zip(xs, losses))
    svg = f"<svg width='{w}' height='{h}' xmlns='http://www.w3.org/2000/svg'><line x1='{m}' y1='{h-m}' x2='{w-m}' y2='{h-m}' stroke='black'/><line x1='{m}' y1='{m}' x2='{m}' y2='{h-m}' stroke='black'/><polyline fill='none' stroke='black' stroke-width='2' points='{pts}'/></svg>"
    b64 = base64.b64encode(svg.encode("utf-8")).decode("utf-8")
    with open(loss_plot.path, "w") as f:
        f.write(f"### Training loss per epoch\n\n<img src='data:image/svg+xml;base64,{b64}'/>")
    os.makedirs(model.path, exist_ok=True)
    joblib.dump(clf, os.path.join(model.path, "model.joblib"))

@dsl.component(base_image=TRAIN_IMAGE)
def deploy_kserve(endpoint_name: str, model: Input[Model]):
    import os, json, time, requests
    with open("/var/run/secrets/kubernetes.io/serviceaccount/namespace","r") as f:
        namespace = f.read().strip()
    with open("/var/run/secrets/kubernetes.io/serviceaccount/token","r") as f:
        token = f.read().strip()
    ca = "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
    def to_kserve_uri(u: str) -> str:
        return u.replace("minio://", "s3://", 1) if u.startswith("minio://") else u
    storage_uri = to_kserve_uri(model.uri)
    host = os.environ.get("KUBERNETES_SERVICE_HOST")
    port = os.environ.get("KUBERNETES_SERVICE_PORT", "443")
    base = f"https://{host}:{port}"
    headers = {"Authorization": f"Bearer {token}", "Content-Type": "application/json"}
    body = {
        "apiVersion": "serving.kserve.io/v1beta1",
        "kind": "InferenceService",
        "metadata": {"name": endpoint_name, "namespace": namespace},
        "spec": {"predictor": {"model": {"modelFormat": {"name": "sklearn"}, "storageUri": storage_uri}}}
    }
    get_url = f"{base}/apis/serving.kserve.io/v1beta1/namespaces/{namespace}/inferenceservices/{endpoint_name}"
    post_url = f"{base}/apis/serving.kserve.io/v1beta1/namespaces/{namespace}/inferenceservices"
    r = requests.get(get_url, headers=headers, verify=ca)
    if r.status_code == 404:
        requests.post(post_url, headers=headers, data=json.dumps(body), verify=ca).raise_for_status()
    else:
        headers_patch = headers.copy()
        headers_patch["Content-Type"] = "application/merge-patch+json"
        requests.patch(get_url, headers=headers_patch, data=json.dumps(body), verify=ca).raise_for_status()
    t0 = time.time()
    url = None
    while time.time() - t0 < 600:
        s = requests.get(get_url, headers=headers, verify=ca).json()
        conds = (s.get("status") or {}).get("conditions") or []
        ready = any(c.get("type") == "Ready" and str(c.get("status")).lower() == "true" for c in conds)
        url = (s.get("status") or {}).get("url")
        if ready and url:
            break
        time.sleep(5)
    print({"namespace": namespace, "name": endpoint_name, "url": url, "storageUri": storage_uri})

@dsl.pipeline(name="iris-train-and-serve-artifacts")
def pipeline(epochs: int = 20, endpoint_name: str = f"iris-sgd-{uuid.uuid4().hex[:6]}") -> Model:
    t = train_and_log(epochs=epochs)
    deploy_kserve(endpoint_name=endpoint_name, model=t.outputs["model"]).after(t)
    return t.outputs["model"]

client = kfp.Client()
run = client.create_run_from_pipeline_func(
    pipeline,
    arguments={"epochs": 12, "endpoint_name": f"iris-sgd-{uuid.uuid4().hex[:6]}"},
    experiment_name="demo-artifacts-kserve",
    # pipeline_root="minio://mlpipeline/v2/artifacts"
)
run`
---

Impacted by this bug? Give it a 👍.
